src/readonlyramdb.cpp

```diff
@@ -72,44 +82,71 @@ static auto Measure(const std::string& funcName, TFunc&& func)
 }
 
-void ReadonlyRamDb::Prefetch(const std::string& path, bool warn_if_unreadable)
+void ReadonlyRamDb::ParseAndLoadDb(std::istream& input_stream,
+                                   const std::string& path,
+                                   bool warn_if_unreadable)
 {
-    Measure("Prefetch", [this, &path, warn_if_unreadable]() {
-        auto file = std::ifstream{path};
+    if(!input_stream)
+    {
+        const auto log_level = (warn_if_unreadable && !MIOPEN_DISABLE_SYSDB) ? LoggingLevel::Warning
+                                                                             : LoggingLevel::Info;
+        MIOPEN_LOG(log_level, "File is unreadable: " << path);
+        return;
+    }
 
-        if(!file)
-        {
-            const auto log_level = (warn_if_unreadable && !MIOPEN_DISABLE_SYSDB)
-                                       ? LoggingLevel::Warning
-                                       : LoggingLevel::Info;
-            MIOPEN_LOG(log_level, "File is unreadable: " << path);
-            return;
-        }
+    auto line   = std::string{};
+    auto n_line = 0;
 
-        auto line   = std::string{};
-        auto n_line = 0;
+    while(std::getline(input_stream, line))
+    {
+        ++n_line;
 
-        while(std::getline(file, line))
-        {
-            ++n_line;
+        if(line.empty())
+            continue;
 
-            if(line.empty())
-                continue;
+        const auto key_size = line.find('=');
+        const bool is_key   = (key_size != std::string::npos && key_size != 0);
+
+        if(!is_key)
+        {
+            MIOPEN_LOG_E("Ill-formed record: key not found: " << path << "#" << n_line);
+            continue;
+        }
 
-            const auto key_size = line.find('=');
-            const bool is_key   = (key_size != std::string::npos && key_size != 0);
+        const auto key      = line.substr(0, key_size);
+        const auto contents = line.substr(key_size + 1);
 
-            if(!is_key)
-            {
-                MIOPEN_LOG_E("Ill-formed record: key not found: " << path << "#" << n_line);
-                continue;
-            }
+        cache.emplace(key, CacheItem{n_line, contents});
+    }
+}
 
-            const auto key      = line.substr(0, key_size);
-            const auto contents = line.substr(key_size + 1);
+void ReadonlyRamDb::Prefetch(const std::string& path, bool warn_if_unreadable)
+{
+    Measure("Prefetch", [this, &path, warn_if_unreadable]() {
 
-            cache.emplace(key, CacheItem{n_line, contents});
+        constexpr bool isEmbedded = MIOPEN_EMBED_DB;
+        if(!testing_find_db_path_override() && isEmbedded)
+        {
+#if MIOPEN_EMBED_DB
+            boost::filesystem::path filepath(path);
+            const auto& it_p = miopen_data().find(filepath.filename().string() + ".o");
+            if(it_p == miopen_data().end())
+                MIOPEN_THROW(miopenStatusInternalError,
```

Comment:
[Suggestion] I think we may want to log an error and just return without preloading anything because it is not a fatal error. But it may as well be left as is because it should not be allowed to happen in production anyway.

Reply:
The later was my reasoning for making it fatal, since MIOpen is compiled specifically to embed the data for a particular architecture, if its not there, something is certainly wrong.